src/sinks/util/rusoto2.rs

```diff
@@ -91,34 +89,15 @@ impl AwsCredentialsProvider {
     }
 }
 
+#[async_trait]
 impl ProvideAwsCredentials for AwsCredentialsProvider {
-    type Future = AwsCredentialsProviderFuture;
-
-    fn credentials(&self) -> Self::Future {
-        match self {
-            Self::Default(p) => AwsCredentialsProviderFuture::Default(p.credentials()),
-            Self::Role(p) => AwsCredentialsProviderFuture::Role(p.credentials()),
-            Self::Static(p) => AwsCredentialsProviderFuture::Static(p.credentials()),
-        }
-    }
-}
-
-pub enum AwsCredentialsProviderFuture {
-    Default(AutoRefreshingProviderFuture<ChainProvider>),
-    Role(AutoRefreshingProviderFuture<StsAssumeRoleSessionCredentialsProvider>),
-    Static(FutureResult<AwsCredentials, CredentialsError>),
-}
-
-impl Future01 for AwsCredentialsProviderFuture {
-    type Item = AwsCredentials;
-    type Error = CredentialsError;
-
-    fn poll(&mut self) -> Poll01<Self::Item, Self::Error> {
-        match self {
-            Self::Default(f) => f.poll(),
-            Self::Role(f) => f.poll(),
-            Self::Static(f) => f.poll(),
-        }
+    async fn credentials(&self) -> Result<AwsCredentials, CredentialsError> {
+        let fut = match self {
+            AwsCredentialsProvider::Default(p) => p.credentials(),
+            AwsCredentialsProvider::Role(p) => p.credentials(),
+            AwsCredentialsProvider::Static(p) => p.credentials(),
```

Comment:
Can these use `Self` instead of spelling out `AwsCredentialsProvider` each time or does that cause problems with `async_trait`?

Reply:
It was problem in `async_trait`, I think that it was fixed in 1.29 so I update to latest version.
https://github.com/dtolnay/async-trait/search?q=e0401&type=Issues